src/refs.c

```diff
@@ -187,9 +190,29 @@ static int reference_normalize_for_repo(
 		precompose)
 		flags |= GIT_REF_FORMAT__PRECOMPOSE_UNICODE;
 
+	if (!validate) {
```

Comment:
Stylistic nitpick: we don't put braces around single lines (unless there was another multiline condition).  Eg, we would prefer this as:

    if (!validate)
        flags |= GIT_REF_VALIDATION_DISABLE;